### ISSUE TITLE
Prompt not visible in 3.1 if non-ascii character is used

### ISSUE THREAD

**Issue by jorgelt** [fish-shell/fish-shell#6736]
Fish 3.1.0 in Arch Linux using Konsole (but also reproducible in Xterm and Ctrl+Alt+f1/f6 virtual console) with $TERM = xterm-256color.

The prompt is not visible anymore just by changing the prompt to "Sorin" using fish_config. I can still type commands but the prompt is not visible. I observed that the problem happens as soon as I use a non-ascii character in fish_prompt. In the example visible in the screenshot I use "❯" from the Sorin prompt. Also, if I execute "fish_prompt" the expected prompt is printed:
![image](https://user-images.githubusercontent.com/45671419/76440821-b59b2080-63be-11ea-855d-83b32f406703.png)

I believe it is probably not related to fonts since:
- The prompt is printed if I manually type fish_prompt
- Spacefish was working fine with fish 3.0
- Starship works fine in bash

How can I debug this? Any tip?

**Comment by faho** [comment]
Please give the output of `locale`

**Comment by jorgelt** [comment]
> locale
LANG=C
LC_CTYPE="C"
LC_NUMERIC="C"
LC_TIME="C"
LC_COLLATE="C"
LC_MONETARY="C"
LC_MESSAGES="C"
LC_PAPER="C"
LC_NAME="C"
LC_ADDRESS="C"
LC_TELEPHONE="C"
LC_MEASUREMENT="C"
LC_IDENTIFICATION="C"
LC_ALL=

**Comment by faho** [comment]
That's your issue. Please set the locale to something that can handle unicode, like one of the ".UTF-8" locales.

See https://wiki.archlinux.org/index.php/Locale. Most likely you need to generate one, then configure it in /etc/locale.conf.

-----

This is still not a great failure mode - what it's supposed to do is to replace the unknown characters with "?", which is also what happens in my prompt, but for some reason the sorin prompt's

```fish
function fish_prompt; echo -n (set_color red)'❯'(set_color yellow)'❯'(set_color green)'❯ '; end
```

doesn't.

**Comment by jorgelt** [comment]
Thank you @faho indeed that was the issue. Changing the locale to en_US.UTF-8 solved it.

Should I close the issue or do you prefer to keep it open to track the failure mode not changing the unknown characters with "?" ?

**Comment by faho** [comment]
Let's keep it open, I'm bisecting.

**Comment by faho** [comment]
Bisects to 5134949a1. Well, I say that. Unfortunately it had a crash bug with a C locale which was only fixed by 2b0b3d319, but cherry-picking that on top makes it not crash yet fail to output.

cc @ridiculousfish Mind taking a look? The issue is that

```fish
function fish_prompt; echo 😃; end
```

or other non-ascii chars causes no visible output with a C locale.

**Comment by faho** [comment]
Okay, it's actually 2b0b3d3.

**Comment by faho** [comment]
Fixed in master with 6237a24573f0b44b19d4cc7c2def92a221069337, 3.1.1 with 1a063fe3c.

(I forgot to include the magical "fixes" comment)
